### ISSUE TITLE
LibJS: Numeric string keys with leading zeros should not be treated as numeric properties

### ISSUE THREAD

**Issue by linusg** [SerenityOS/serenity#4037]
```
> o = {"1": 1, "01": 2, "001": 3}
{ "1": 3 }
> a = []
[  ]
> a["00"] = "foo"
"foo"
> a[0] === "foo"
true
```

**Comment by davidot** [comment]
I believe this was fixed in f7e2994f35ec8214cffddec802c4cf9f4bd70177.
(Which also fixed "+0", "-0" etc.)

**Comment by sin-ack** [comment]
This issue and many others seem to arise from the fact that `PropertyName` treats things as numbers when it really shouldn't, due to using `StringUtils::convert_to_int`. Perhaps we should write a dead simple number parser for `PropertyName` that doesn't do anything other than convert base 10 numbers.
